### ISSUE TITLE
Loopback 4 don´t close test with a MongoDB and Mocha (V.6.1.0)

### ISSUE THREAD

**Issue by jotamora** [strongloop/loopback-next#3672]
Hello everyone,

I Launch npm test, but it never end. It happens when there is a connection with a MongoDB involve. It looks like loopback don´t close the conections: 

```
before('setupApplication', async () => {
        ({ app, client } = await setupApplication());
        //await prepareTestData()
    });

    after(async () => {
        //await cleanupTestData();
        await app.stop();
    });

    it('login with a test user', async () => {
        const res = await client.post('/user-accounts/login').set('urlLogin', TEST_TENANT_URL)
            .send({
                email: TEST_EMAIL,
                password: TEST_PASS
            }).expect(200);
        token = res.body.token;
        expect(res.body).to.ownProperty("token").and.length(EXPECTED_TOKEN_LENGTH);
        console.log("Logged in with token ", res.body.token);
    });
}
```
Result: The test in the terminal never end. 

Mocha is waiting because the "mongodb" package has pending tasks. In fact, I tried to stop the MongoDB service and indeed after a few seconds (I understand that due to timeout), the tests are over. 

I have try to close the "datasource" (calling "disconnect ()" after a period of time) but the tests still don't end. I have try app.stop() and app.close() too, but it not works. 

What can I do? Don´t should do loopback it automatically? 

Any help would be great.

Thanks.

## Acceptance Criteria
_Extracted from @bajtos' [comment below](https://github.com/strongloop/loopback-next/issues/3672#issuecomment-528742343)_
- [ ] apply this change to our project template and add exit option to the default .mocharc file - see the template here:
https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/project/templates/.mocharc.json

**Comment by bajtos** [comment]
Please use Mocha option `exit`, it will end the process after all tests (and after/afterEach hooks) have finished. See https://mochajs.org/#configuring-mocha-nodejs to learn more. Alternatively, you can add `--exit` option to `mocha` CLI, e.g. `mocha --exit src/__tests__/**/*.js`.

IMO, we should apply this change to our project template and add `exit` option to the default `.mocharc` file - see the template here:

https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/project/templates/.mocharc.json

In the longer term, I think we should improve `app.stop()` to close all datasources too. I opened a new issue to discuss that change - see https://github.com/strongloop/loopback-next/issues/3675

**Comment by bajtos** [comment]
> IMO, we should apply this change to our project template and add `exit` option to the default `.mocharc` file - see the template here:
> 
> https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/project/templates/.mocharc.json

See also https://github.com/strongloop/loopback-next/issues/2403

**Comment by rbarriuso** [comment]
@jotamora Perhaps this works: https://github.com/strongloop/loopback-next/issues/3675#issuecomment-528886582

**Comment by jotamora** [comment]
@rbarriuso I have tried, it still not working.

**Comment by bajtos** [comment]
The problem should have been fixed by #3678. Feel free to re-open the issue (or leave a comment) if the problem is still present, e.g. `app.stop()` is not closing the MongoDB connection and thus Mocha does not exit.
